src/components/Loader/index.js

```diff
@@ -110,20 +110,19 @@ export function useLoader(fetch, updateProps = []) {
     [fetch, state]
   );
 
-  React.useEffect(
-    () => {
+  const refreshDataCallback = React.useCallback(
+    function incrementAndFetchData() {
       mounted.current += 1;
       fetchDataCallback(mounted.current);
     },
-    updateProps // eslint-disable-line react-hooks/exhaustive-deps
+    [fetchDataCallback]
   );
 
+  React.useEffect(refreshDataCallback, updateProps);
```

Comment:
Very interesting solution! Thanks for contributing it! I ran [several tests](https://github.com/AlexsLemonade/refinebio-frontend/blob/320c708730cee1d0e353f2b3f200dc00396fdbae/src/components/Loader/loader.test.js) and this passed all of them.

It seems eslint complains if you use this syntax:

```js
React.useEffect(() => refreshDataCallback(), updateProps);
```

Why do you think this happens? I'm wondering if the original version has the same issues but is a case that the rule can't detect.

Reply:
Hey, I think the fact that it throws a warning makes sense... however the specific warning that 'updateProps' is not an array... I haven't made any sense of that.

Reply:
I get two warnings:

> [eslint] React Hook React.useEffect was passed a dependency list that is not an array literal. This means we can't statically verify whether you've passed the correct dependencies. (react-hooks/exhaustive-deps)

I don't think there's any way around this one, since the idea is to have the effect run whenever the values of `updateProps` change, and that must be a parameter :/

The one that worried me more was:

> [eslint] React Hook React.useEffect has a missing dependency: 'refreshDataCallback'. Either include it or remove the dependency array. (react-hooks/exhaustive-deps)

Seems that the recommendation is to always include [functions in the list of effect dependencies](https://overreacted.io/a-complete-guide-to-useeffect/#moving-functions-inside-effects).

I think we should merge this one since it's better than the current solution in `dev`, and see if #651 makes any difference.